tests/test_wellbore_marker_frame.py

```diff
@@ -0,0 +1,433 @@
+                                                   feature_name = 'horizon_feature_1')
+    horizon_feature_1.create_xml()
+    horizon_interp_1 = rqo.HorizonInterpretation(parent_model = model,
+                                                 title = 'horizon_interp_1',
+                                                 genetic_boundary_feature = horizon_feature_1,
+                                                 sequence_stratigraphy_surface = 'flooding',
+                                                 boundary_relation_list = ['conformable'])
+    horizon_interp_1.create_xml()
+
+    horizon_feature_2 = rqo.GeneticBoundaryFeature(parent_model = model,
+                                                   kind = 'horizon',
+                                                   feature_name = 'horizon_feature_2')
+    horizon_feature_2.create_xml()
+    horizon_interp_2 = rqo.HorizonInterpretation(parent_model = model,
+                                                 title = 'horizon_interp_2',
+                                                 genetic_boundary_feature = horizon_feature_1,
+                                                 sequence_stratigraphy_surface = 'transgressive',
+                                                 boundary_relation_list = ['conformable'])
+    horizon_interp_2.create_xml()
+
+    fault_feature_1 = rqo.TectonicBoundaryFeature(parent_model = model,
+                                                  kind = 'fault',
+                                                  feature_name = 'fault_feature_1')
+    fault_feature_1.create_xml()
+    fault_interp_1 = rqo.FaultInterpretation(parent_model = model,
+                                             title = 'fault_interp_1',
+                                             tectonic_boundary_feature = fault_feature_1,
+                                             is_normal = True,
+                                             maximum_throw = 15)
+    fault_interp_1.create_xml()
+
+    wellbore_marker_frame_dataframe = pd.DataFrame({
+        'MD': [300, 310, 330],
+        'Type': ['horizon', 'horizon', 'fault'],
+        'Surface': ['marker_horizon_1', 'marker_horizon_2', 'marker_fault_1'],
+        'Interp_Surface': ['horizon_interp_1', 'horizon_interp_2', 'fault_interp_1'],
+        'Well': well_name
+    })
+
+    # Create a wellbore marker frame from a dataframe
+    wellbore_marker_frame = resqpy.well.WellboreMarkerFrame.from_dataframe(parent_model = model,
+                                                                           dataframe = wellbore_marker_frame_dataframe,
+                                                                           trajectory = trajectory)
+    print(wellbore_marker_frame.wellbore_marker_list)
+    # --------- Act ----------
+    # Get the marker object from an interpretation object and interpretation object uuid
+    found_marker1 = wellbore_marker_frame.find_marker_from_index(1)
```

Comment:
If I recall, this method is due to be changed from 1 based indexing to pythonic zero based – which shouldn't affect the result of this test as it is written. However, it might be better to make the assertion(s) specific to the current implementation of the method, so that it will fail when we change the definition of the method. That way, we will be more sure that we have made the change as we intended. (Of course we will have to change the test to get it to pass again, once we've seen it fail.)

Reply:
I changed the assertion statement to:
assert found_marker1[3] == wellbore_marker_frame['Type'][2]. The test will fail once we change the indexing in the method to zero-based